README.md

```diff
@@ -79,23 +79,28 @@ package are mentioned in the `package/snapper.changes` file, update also the
 - Make sure the units tests still passes ([see above](#running-tests)).
 
 - When the version is increased then the Git repo has to be tagged, use the
-`vX.Y.Z` format for the tag. Also the
-[filesystems:snapper](https://build.opensuse.org/project/show/filesystems:snapper)
-OBS project has to be updated.
-
-- To create the package use command `make package`. Then use the common work-flow to submit
-package to the build service. For [openSUSE:Factory](https://build.opensuse.org/project/show/openSUSE:Factory)
-send at first the package to the devel project
-[YaST:Head](https://build.opensuse.org/project/show/YaST:Head) in OBS.  
-*Please note that this OBS project builds for more distributions so more metadata
-files have to be updated. See the OBS documentation for more info ([cross distribution
-howto](https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto),
-[Debian builds](https://en.opensuse.org/openSUSE:Build_Service_Debian_builds)).*
+  `vX.Y.Z` format for the tag. Also the [filesystems:snapper][]
+  OBS project has to be updated.
+
+- To create the package use command `make package`. Then use the common
+  work-flow to submit the
+package to the build service. For [openSUSE:Factory][]
+  send at first the package to the devel project [filesystems:snapper][] in OBS.
+
+    *Please note that this OBS project builds for more distributions
+    so more metadata files have to be updated. See the OBS documentation
+    for more info ([cross distribution howto][xdist], [Debian builds][]).*
```

Comment:
@aschnell BTW I have checked the build targets in https://build.opensuse.org/package/show/filesystems:snapper/snapper and it includes several xUbuntu versions that are long out of support. How about leaving only the LTS ones, 16.04, 18.04 + newer?

<details>
<summary>Ubuntu_version_history</summary>
https://en.wikipedia.org/wiki/Ubuntu_version_history
"The support period for non-LTS releases is 9 months."

Ubuntu 14.04 LTS (Trusty Tahr)
"Normal LTS support is set to continue until 25 April 2019, after which extended security maintenance will be available to Ubuntu Advantage customers and as a separate commercial purchase, as was the case previously with 12.04."

Ubuntu 16.04 LTS (Xenial Xerus)
Ubuntu 18.04 LTS (Bionic Beaver)

https://xubuntu.org/releases/ : "Regular Xubuntu releases are supported for 9 months and the LTS releases for 3 years."
</details>

Similarly it includes oS 13.1 which is 3 years out of support: https://en.opensuse.org/Lifetime

Fedora 20 is 4 years out of support: https://fedoraproject.org/wiki/End_of_life

Reply:
Since the build service still provides those versions it seems as if somebody sees a use case.
On the other hand, what is gained for snapper by removing them? Can snapper use newer language or library features to simplify code?

Reply:
Well, there *is* a 6 year old piece of conditional compilation that we could remove:
https://github.com/openSUSE/snapper/blob/ec58ba76a4f86405184b0ba4c6c2fa274a7c0469/snapper/Btrfs.cc#L1371-L1392

OTOH you're right, we should not fix what ain't broken. IMHO we should touch it only once something breaks. I just got curious seeing so many build targets.

Reply:
That code could indeed be removed then. When it comes to threads it might be worth to switch from boost to C++11. Same for regex.